README.md

```diff
@@ -18,211 +19,195 @@
 ```
 
-### Rows and Data
-
-Both `simpleQuery` and `query` return the same `PostgresRow` type. Columns can be fetched from the row using the `column(_: String)` method.
+The query will return a [`PostgresRowSequence`], which is an AsyncSequence of [`PostgresRow`]s. You can 
+consume the rows 1 by 1: 
 
 ```swift
-let row: PostgresRow = ...
-let version = row.column("version")
-print(version) // PostgresData?
+for try await row in rows {
+  // do something with the row
+}
 ```
 
-`PostgresRow` columns are stored as `PostgresData`. This struct contains the raw bytes returned by PostgreSQL as well as some information for parsing them, such as:
-
-- Postgres column type
-- Wire format: binary or text
-- Value as array of bytes
+#### Decoding from PostgresRow
 
-`PostgresData` has a variety of convenience methods for converting column data to usable Swift types.
+However in most cases you want to cast a row directly into Swift primitive datatypes:
 
 ```swift
-let data: PostgresData= ...
+for try await (id, username, birthday) in rows.decode((Int, String, Date).self, context: .default) {
+  // do something with the datatypes.
+}
+```
+
+To be able to decode a Postgres value to a Swift struct, the type needs to implement the 
+`PostgresDecodable` protocol. PostgresNIO has default implementations for a number of Swift 
+primitives:
 
-print(data.string) // String?
+- Bool
+- Bytes, Data, ByteBuffer
+- Date
+- UInt8, Int16, Int32, Int64, Int, Float, Double
+- String
+- UUID
 
-// Postgres only supports signed Ints.
-print(data.int) // Int?
-print(data.int16) // Int16?
-print(data.int32) // Int32?
-print(data.int64) // Int64?
+#### Quering with parameters
 
-// 'char' can be interpreted as a UInt8. 
-// It will show in db as a character though. 
-print(data.uint8) // UInt8?
+Sending parameterized queries to the database is also supported. (In the coolest way possible):
 
-print(data.bool) // Bool?
+```swift
+let id = 1
+let username = "fancyuser"
+let birthday = Date()
+try await connection.query("""
+  INSERT INTO users (id, username, birthday) VALUES (\(id), \(username), \(birthday))
+  """, 
+  logger: logger
+)
+```
 
-print(try data.jsonb(as: Foo.self)) // Foo?
+Yes, that looks like [SQL injection](https://en.wikipedia.org/wiki/SQL_injection) 😱, but this is 
+actually save in PostgresNIO. The first parameter, of the query function is not a String, but a 
+PostgresQuery, that implements the `ExpressibleByStringInterpolation` protocol. For this reason, 
+PostgresNIO uses the static string as the SQL and the interpolation parameters as the bindings of 
+the query. To interpolate values into the query, your values need to implement the 
+`PostgresEncodable` protocol. 
 
-print(data.float) // Float?
-print(data.double) // Double?
+For some queries you won't receive any rows from the server (mainly `INSERT`, `UPDATE` and `DELETE`). 
+To make the API support this, the result of the `query` function is an `@discardableResult`. 
```

Comment:
```suggestion
Some queries do not receive any rows from the server (most often `INSERT`, `UPDATE`, and `DELETE` queries with no `RETURNING` clause, not to mention most DDL queries). To support this, the `query(_:logger:)` method is marked `@discardableResult`, so that the compiler does not issue a warning if the return value is not used. 
```